qiskit/algorithms/time_evolvers/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,178 @@
+from qiskit.algorithms.observables_evaluator import eval_observables
+from qiskit.algorithms.time_evolvers.evolution_problem import EvolutionProblem
+from qiskit.algorithms.time_evolvers.evolution_result import EvolutionResult
+from qiskit.algorithms.time_evolvers.real_evolver import RealEvolver
+from qiskit.opflow import (
+    SummedOp,
+    PauliOp,
+    CircuitOp,
+    PauliSumOp,
+    StateFn,
+    OperatorBase,
+)
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.primitives import Estimator
+from qiskit.synthesis import ProductFormula, LieTrotter
+
+
+class TrotterQRTE(RealEvolver):
+    """Quantum Real Time Evolution using Trotterization.
+    Type of Trotterization is defined by a ProductFormula provided.
+
+    Examples:
+
+        .. jupyter-execute::
+
+            from qiskit.opflow import X, Z, Zero
+            from qiskit.algorithms import EvolutionProblem, TrotterQRTE
+
+            operator = X + Z
+            initial_state = Zero
+            time = 1
+            evolution_problem = EvolutionProblem(operator, 1, initial_state)
+            # LieTrotter with 1 rep
+            trotter_qrte = TrotterQRTE()
+            evolved_state = trotter_qrte.evolve(evolution_problem).evolved_state
+    """
+
+    def __init__(
+        self,
+        product_formula: Optional[ProductFormula] = None,
+        estimator: Estimator = None,
+    ) -> None:
+        """
+        Args:
+            product_formula: A Lie-Trotter-Suzuki product formula. The default is the Lie-Trotter
+                first order product formula with a single repetition.
+            estimator: An estimator primitive used for calculating expectation values of
+                EvolutionProblem.aux_operators.
+        """
+        if product_formula is None:
+            product_formula = LieTrotter()
+        self.product_formula = product_formula
+        self.estimator = estimator
+
+    @classmethod
+    def supports_aux_operators(cls) -> bool:
+        """
+        Whether computing the expectation value of auxiliary operators is supported.
+
+        Returns:
+            True if ``aux_operators`` expectations in the EvolutionProblem can be evaluated, False
+                otherwise.
+        """
+        return True
+
+    def evolve(self, evolution_problem: EvolutionProblem) -> EvolutionResult:
+        """
+        Evolves a quantum state for a given time using the Trotterization method
+        based on a product formula provided. The result is provided in the form of a quantum
+        circuit. If auxiliary operators are included in the ``evolution_problem``, they are
+        evaluated on an evolved state using an estimator primitive provided.
+
+        .. note::
+            Time-dependent Hamiltonians are not yet supported.
+
+        Args:
+            evolution_problem: Instance defining evolution problem. For the included Hamiltonian,
+                ``PauliOp``, ``SummedOp`` or ``PauliSumOp`` are supported by TrotterQRTE.
```

Comment:
We can always convert any `BaseOperator` to a `SparsePauliOp` I think
```suggestion
            evolution_problem: Instance defining evolution problem.
```